src/app/services/speechtotext/speech_to_text.py

```diff
@@ -0,0 +1,223 @@
+import io
+import os
+from enum import Enum, unique, auto
+
+from PyQt5.QtCore import QObject, pyqtSignal, pyqtSlot
+
+from google.cloud import speech
+
+
+@unique
+class EncodingTypes(Enum):
+    ENCODING_UNSPECIFIED = 'ENCODING_UNSPECIFIED'
+    FLAC = 'FLAC'
+    AMR = 'AMR'
+    AMR_WB = 'AMR_WB'
+    LINEAR16 = 'LINEAR16'
+    OGG_OPUS = 'OGG_OPUS'
+    SPEEX_WITH_HEADER_BYTE = 'SPEEX_WITH_HEADER_BYTE'
+
+
+@unique
+class LanguageCodes(Enum):
+    FR_CA = 'fr-CA'
+    EN_CA = 'en-CA'
+
+
+@unique
+class Models(Enum):
+    DEFAULT = 'default'
+    COMMAND_AND_SEARCH = 'command_and_search'
+    PHONE_CALL = 'phone_call'
+    VIDEO = 'video' 
+
+
+class SpeechToText(QObject):
+
+    transcriptionReady = pyqtSignal(str)
+    exception = pyqtSignal(Exception)
+
+    # All the parameters needed to perform a transcription.
+    __config = {'audioDataPath' : '',
+                'encoding' : None,
+                'enhanced' : False,
+                'languageCode' : None,
+                'model' : None,
+                'outputFolder' : '',
+                'sampleRate' : 0,
+                'serviceAccountPath' : ''}
```

Comment:
Would be better like this:

    config = {
        'audioDataPath' : self.audioDataPath.text(),
        'encoding' : self.encoding.currentText(),
        'enhanced' : self.enhanced.checkState(),
        'languageCode' : self.language.currentText(),
        'model' : self.model.currentText(),
        'outputFolder' : self.window().getSetting('defaultOutputFolder'),
        'sampleRate' : self.sampleRate.value(),
        'serviceAccountPath' : self.window().getSetting('serviceAccountPath')
    }